Sources/ContainerizationOCI/Client/LocalOCILayoutClient.swift

```
@@ -53,7 +54,34 @@ package final class LocalOCILayoutClient: ContentClient {
                     ProgressEvent(event: "add-size", value: fileSize)
                 ])
             }
+        } catch let error as NSError {
+            guard error.code == NSFileWriteFileExistsError else {
+                throw error
+            }
+
+            do {
+                let expectedDigest = try c.digest()
+                let existingData = try Data(contentsOf: file)
```

We may have issues for large files as this loads the entire file into our memory.

---

We read a file only when we attempt to write data into a file that already exists. This allows to determine whether we should throw an error or treat it as already existing data, which requires this verification.

---

@adityaramani Please confirm that we can't throw an error any time the file already exists.

---

If I craft an image with file `/foo` which is 10gb, and then I create another layer that has a file `/foo`, we will hit this branch and cause the code to load the existing 10gb worth of `/foo` into memory?

---

Done